File: lintcode/single-number.py
#!/usr/bin/env python
# encoding: utf-8

# Question: http://www.lintcode.com/en/problem/single-number/

class Solution:
    """
    @param A : an integer array
    @return : a integer
    """
    def singleNumber(self, A):
        # write your code here
        res = 0
        for n in A:
            res ^= n
        return res

File: leetcode/single-number.cc
// Question: https://leetcode.com/problems/single-number/

#include "headers.h"

class Solution {
public:
    int singleNumber(vector<int>& nums) {
        int ans = 0;
        for (auto n : nums) {
            ans ^= n;
        }
        return ans;
    }
};

int main(int argc, char** argv) {
    Solution s;
    vector<int> nums{1,2,3,4,5,1,2,3,4};
    cout << s.singleNumber(nums) << endl;
    return 0;
}
